src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -56,6 +59,17 @@ public HttpsConnectionMiddleware(ConnectionDelegate next, HttpsConnectionAdapter
                 }
             }
 
+            if (options.HttpProtocols == HttpProtocols.Http1AndHttp2)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                    && Environment.OSVersion.Version >= new Version(6, 2)
+                    && Environment.OSVersion.Version < new Version(10, 0))
+                {
+                    _logger.LogWarning(4, CoreStrings.HTTP2DefaultCiphersInsufficient);
```

Comment:
😕 This seems rather awkward, assigning LogIds manually. Shouldn't this be moved to a Log class?